op-service/backoff/strategies.go

```diff
@@ -17,23 +17,23 @@ type Strategy interface {
 // function is min(e.Min + (2^attempt * 1000) + randBetween(0, e.MaxJitter), e.Max)
 type ExponentialStrategy struct {
 	// Min is the minimum amount of time to wait between attempts in ms.
-	Min float64
+	Min time.Duration
 
 	// Max is the maximum amount of time to wait between attempts in ms.
-	Max float64
+	Max time.Duration
 
 	// MaxJitter is the maximum amount of random jitter to insert between
 	// attempts in ms.
-	MaxJitter int
+	MaxJitter time.Duration
 }
 
 func (e *ExponentialStrategy) Duration(attempt int) time.Duration {
-	var jitter int
+	var jitter time.Duration
 	if e.MaxJitter > 0 {
-		jitter = rand.Intn(e.MaxJitter)
+		jitter = time.Duration(rand.Int63n(e.MaxJitter.Milliseconds()))
 	}
-	dur := e.Min + (math.Pow(2, float64(attempt)) * 1000)
-	dur += float64(jitter)
+	dur := e.Min + time.Duration((math.Pow(2, float64(attempt)) * 1000))
+	dur += jitter
```

Comment:
Also needs corrections. You can't just convert from int to time.Duration, it has to scale with the correct unit.

Reply:
Got it